src/amo/components/Category/index.js

```diff
@@ -67,11 +68,17 @@ export function mapStateToProps(state, ownProps) {
     { ...state.search.filters, page: parsePage(state.search.page) },
     { ...filters, page: queryParams.page },
   );
+
+  const hasFetched = state.categories.hasFetched;
+  const loading = state.categories.loading || state.search.loading;
+
   if (filtersMatchState) {
     return {
       addonType: filters.addonType,
       category,
       filters,
+      hasFetched,
+      loading,
       pathname,
       queryParams,
       ...state.search,
```

Comment:
I know this is old code but could you remove the spread and instead reference each one of the `state` attributes?